src/nlsolve/anderson.jl

```diff
@@ -90,12 +90,16 @@ function (S::NLAnderson{true})(integrator)
   # precalculations
   κtol = κ*tol
 
-  zs = zeros(length(vec(z)), S.n+1)
-  gs = zeros(length(vec(z)), S.n+1)
+  zs = fill(Any[], S.n+1)
```

Comment:
this should fill with `typeof(u)`. In fact, I would just do the comprehension `zs = [similar(u) for i in 1:S.n+1]`